test/pylib/resource_gather.py

```diff
@@ -0,0 +1,195 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+
+import asyncio
+import getpass
+import os
+import platform
+import subprocess
+from abc import ABC
+from asyncio import Task, Event
+from datetime import datetime
+from pathlib import Path
+from typing import TextIO
+
+import psutil
+
+from test.pylib.db.model import Metric, SystemResourceMetric, CgroupMetric, Test
+from test.pylib.db.writer import DATE_TIME_TEMPLATE, SQLiteWriter, SYSTEM_RESOURCE_METRICS_TABLE, METRICS_TABLE, \
+    DEFAULT_DB_NAME, CGROUP_MEMORY_METRICS_TABLE, TESTS_TABLE
+
+CGROUP_INITIAL: Path = Path('/sys/fs/cgroup/initial')
+CGROUP_TESTS: Path = CGROUP_INITIAL.parent / 'tests'
+cancel_event_global = None
+stop_event_global = None
+
+class ResourceGather(ABC):
+    def __init__(self, test, tmp_dir: str):
+        self.test = test
+        self.db_path = Path(tmp_dir) / DEFAULT_DB_NAME
+        standardized_name = self.test.shortname.replace("/", "_")
+        self.cgroup_path = Path(
+            f"{CGROUP_TESTS}/{self.test.suite.name}.{standardized_name}.{self.test.suite.mode}.{self.test.id}"
+        )
+
+    def make_cgroup(self):
+        pass
+
+    def put_process_to_cgroup(self):
+        os.setsid()
+
+    def get_test_metrics(self) -> Metric:
+        pass
+
+    def write_metrics_to_db(self, metrics: Metric, success: bool = False) -> None:
+        pass
+
+    def cgroup_monitor(self, test_event: Event):
+        pass
+
+
+class ResourceGatherOff(ResourceGather):
+    def cgroup_monitor(self, test_event) -> Task:
+        return asyncio.create_task(no_monitor())
+
+
+class ResourceGatherOn(ResourceGather):
+    def __init__(self, test, tmp_dir: str):
+        super().__init__(test, tmp_dir)
+        self.sqlite_writer = SQLiteWriter(self.db_path)
+        self.test_id: int = self.sqlite_writer.write_row_if_not_exist(
+            Test(
+                architecture=platform.machine(),
+                directory=test.suite.name,
+                mode=test.mode,
+                run_id=test.id,
+                test_name=test.shortname
+            ),
+            TESTS_TABLE)
+
+    def make_cgroup(self):
+        os.makedirs(self.cgroup_path, exist_ok=True)
+
+    def get_test_metrics(self) -> Metric:
+        test_metrics: Metric = Metric(test_id=self.test_id)
+        test_metrics.time_taken = round(int(self.test.time_end - self.test.time_start))
```

Comment:
Why round?

Reply:
Missed to change it. It's redundant, so I'll remove it.